modules/S09_firmware_base_version_check.sh

```diff
@@ -154,8 +154,13 @@ S09_firmware_base_version_check() {
     local lTMP_PID="$!"
     store_kill_pids "${lTMP_PID}"
     WAIT_PIDS_S09_1+=( "${lTMP_PID}" )
-    max_pids_protection $(( "${MAX_MOD_THREADS}"*3 )) "${WAIT_PIDS_S09_1[@]}"
-  done
+    if [[ "${#WAIT_PIDS_S09_1[@]}" -gt "${MAX_MOD_THREADS}" ]]; then
+      recover_wait_pids WAIT_PIDS_S09_1
```

Comment:
Is this a mix of strings and arrays? In my opinion it should look the following:

```
recover_wait_pids "${WAIT_PIDS_S09_1[@]}"
```

Reply:
found it https://www.gnu.org/software/bash/manual/bash.html#Shell-Parameters